tiledb/__init__.py

```diff
@@ -19,23 +19,6 @@
 else:
     lib_name = "tiledb"
 
-# On Windows and whl builds, we may have a shared library already linked, or
-# adjacent to, the cython .pyd shared object. In this case, we can import directly
-# from .libtiledb
-try:
-    import tiledb
-
-    from .libtiledb import Ctx
-
-    del Ctx
-except:
-    try:
-        lib_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "native")
-        ctypes.CDLL(os.path.join(lib_dir, lib_name))
-    except OSError:
-        # Otherwise try loading by name only.
-        ctypes.CDLL(lib_name)
-
```

Comment:
Not sure if this code is supposed to be removed. @ihnorton there is a reason for this here, right?

Reply:
No, this should not be removed.

Reply:
As far as I know, right now the produced wheel works on Windows, but I don't understand the incentive behind this code snippet. Why do we need it? I will put it back and test everything, but I am curious why it is there.

Reply:
Similar looking code was removed from TileDB-VCF when it was migrated to scikit-build-core:

https://github.com/TileDB-Inc/TileDB-VCF/pull/701/files#diff-08ebe1f990ba71b94e5815abc464c931de9c8ffaba809aaa5768ede105399c9b

Reply:
Ok, please test that editable installs work on Windows with this change -- if they do, then we can proceed.

Reply:
I tested editable installs on Windows, works perfectly! 💯